loader/src/main/java/net/neoforged/fml/common/asm/enumextension/IndexedEnum.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.fml.common.asm.enumextension;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Specifies that the annotated enum has an int ID parameter which must match the enum constant's ordinal
+ */
+@Target(ElementType.TYPE)
+public @interface IndexedEnum {
```

Comment:
Would it be an idea to have this as a parameter level annotation in the ctor?

Reply:
Disregard, this would increase the patch length significantly, making it unreadable.